### PR TITLE
Update Testing

### PR DESCRIPTION
**Pull request by sean-m-sullivan**
# What does this PR do?
Updating testing to add more coverage to fix approval testing for 4.5/4.4
Add back publish role testing, fix publish role to be namespace agnostic like other roles.

### CODE REVIEW

File: roles/publish/tasks/main.yml [hunk 1]
```diff
@@ -29,7 +29,7 @@
     - __ah_collection_item.collection_local_path is defined
 
 - name: Build Collections
-  infra.ah_configuration.ah_build:
```
Comment (djdanielsson) [inline roles/publish/tasks/main.yml:32]: I am fine with this until 2.4 then we should FQCN everything possible to infra

File: .github/tests/playbooks/testing_playbook.yml [hunk 2]
```diff
@@ -36,20 +36,15 @@
...
     - name: Upload Collection
       ansible.builtin.include_role:
         name: collection
 
-    #  - name: Publish Collections
-    #    ansible.builtin.include_role:
-    #      name: publish
-    #    vars:
-    #      ah_auto_approve: true
+    - name: Publish Collections
```
Comment (djdanielsson) [inline .github/tests/playbooks/testing_playbook.yml:43]: Shouldn't this say publish? I'm on mobile so kinda hard to tell 100%

**Review by djdanielsson — APPROVED** [review]
LGTM

File: roles/publish/tasks/main.yml (hunk 1 above)
Comment (sean-m-sullivan) [reply roles/publish/tasks/main.yml:32]: When we switch it over we should switch the testing over to use infra as well, and all the roles at the same time, Agreed, This was more lets make it congruent in the meantime, and was causing testing fail as we were building with redhat_cop.

File: .github/tests/playbooks/testing_playbook.yml (hunk 2 above)
Comment (sean-m-sullivan) [reply .github/tests/playbooks/testing_playbook.yml:43]: it does say publish, thought this was another.

**Pull request merged**